react-backend/client/src/components/visitors/qrcode.js

```diff
@@ -98,6 +98,11 @@ export class QRCode extends Component {
       headers: this.headers,
       body: JSON.stringify(visitInfo),
     })
+      .then(res => {
+        if (res.status === 500) {
+          throw new Error();
+        }
+      })
       .then(this.props.history.push('/visitor/end'))
```

Comment:
put this inside of an anonymous function. needs to be `() => this.props.history.push('/visitor/end')`, otherwise you're just directly calling that function.